jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -214,6 +214,22 @@
 	private static boolean reflectCacheDebug;
 	private static boolean reflectCacheAppOnly = true;
 
+	/*
+	 * This <code>Dummy</code> class is created to tell whether the cached
+	 * class value is initialized or not; e.g. <code>cachedDeclaringClass</code> and <code>cachedEnclosingClass</code> 
+	 *
+	 * */
+	private static final class Dummy {}
```

Comment:
Technically it is possible to do
```
private static final Object dummy = new Object();
Class<?> clz = (Class<?>)dummy;
```
later do a comparison against object `dummy`.
I am fine with current `Dummy` class approach unless Pete says otherwise.